src/backend/distributed/transaction/backend_data.c

```diff
@@ -780,6 +784,85 @@ MarkCitusInitiatedCoordinatorBackend(void)
 }
 
 
+/*
+ * AssignGlobalPID assigns a global process id for the current backend
+ */
+void
+AssignGlobalPID(void)
+{
+	if (MyBackendData->globalPID != 0)
+	{
+		elog(DEBUG1, "a global pid is already assigned to this backend");
+		return;
+	}
+
+	if (!IsCitusInitiatedRemoteBackend())
+	{
+		uint64 globalPID = GenerateGlobalPID();
+
+		SpinLockAcquire(&MyBackendData->mutex);
+		MyBackendData->globalPID = globalPID;
+		SpinLockRelease(&MyBackendData->mutex);
+
+		return;
+	}
+
+	uint64 globalPID = ExtractGlobalPID();
+
+	SpinLockAcquire(&MyBackendData->mutex);
+	MyBackendData->globalPID = globalPID;
+	SpinLockRelease(&MyBackendData->mutex);
+}
+
+
+/*
+ * GetGlobalPID returns the global process id of the current backend
+ */
+uint64
+GetGlobalPID(void)
+{
+	uint64 globalPID = 0;
```

Comment:
what if `AssignGlobalPID` has not been called when this function is called? Should we maybe start the function with `AssignGlobalPID ` ?

Reply:
I'll call the `AssignGlobalPID`.